### ISSUE TITLE
RetroPlay Doesn't Work

### ISSUE THREAD

**Issue by Saroufim** [flathub/tv.kodi.Kodi#41]
RetroPlay doesn't seem to be working. No roms whatsoever are recognized and Libretro all in all does not seem to be available in the flatpak version.

**Comment by Saroufim** [comment]
When rom files are selected, nothing happens at all. I tried GBA and SNES roms. Had no luck with both.

**Comment by wsnipex** [comment]
that's expected, since all kodi game addons are missing: https://github.com/kodi-game/
It's going to be a pain to add them without using kodi's build system, which downloads dependencies during building,

**Comment by tilosp** [comment]
@wsnipex are there plans to bring the binary addons repo to linux?

**Comment by wsnipex** [comment]
they are available on Linux since the start, just not via the add-on repo. Same as with all other binary addons.

**Comment by wsnipex** [comment]
If it would be easy to provide binaries that work on every Linux system, flatpak wouldn't exist

**Comment by tilosp** [comment]
the problem with flatpak extensions is that the kodi addons install files to two locations (`/app/lib/kodi/addons/` and `/app/share/kodi/addons/`). And flatpak extensions get mounted into one directory.





> If it would be easy to provide binaries that work on every Linux system, flatpak wouldn't exist

but shouldn't it be possible to setup a kodi add-on repo that provides binary addons that work on every flatpak install? Then the binary addons could be build outside of flatpak-builder using kodi's build system.

**Comment by wsnipex** [comment]
you would have to build them inside the flatpak SDK...
IMO, the only reasonable way to build the binary addons is to use the kodi build system inside flatpak builder.

**Comment by razzeee** [comment]
RetroPlayer is working, at least with the games we ship

**Comment by inf0rmix** [comment]
is there any progress on this ?
as a ubuntu user which, I have to use flatpak now without any emulator addons.

**Comment by razzeee** [comment]
I'm not aware of anyone looking at this, but as I stated above retroplayer itself works.

**Comment by Jusix** [comment]
I was currently look at how to get Dolphin Emu running with my Flatpak Kodi RetroPlayer. Do I understand it correctly that this ins't possible at all at the moment?

**Comment by garbear** [comment]
Correct, Dolphin requires OpenGL rendering, and as of today that's currently unfinished. You can see the current progress in the tracking issue here: https://github.com/xbmc/xbmc/issues/17743

**Comment by Jusix** [comment]
But isn't the preferred method of Dolphin Vulkan these days?

**Comment by garbear** [comment]
> But isn't the preferred method of Dolphin Vulkan these days?

I believe so, which is even harder for Kodi, because we don't have any Vulkan rendering yet.
